spec/system/foods/new_system_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe 'Foods', type: :system do
+  before :all do
+    driven_by(:rack_test)
+  end
+
+  it 'add new food should create new food' do
+    visit '/users/sign_in'
+    within('#new_user') do
+      fill_in 'Email', with: 'adam@domain.com'
+      fill_in 'Password', with: 'password'
+    end
+    click_button 'Log in'
+    visit foods_path
+
+    click_link('Add Food')
```

Comment:
This is a good test for the index view, but I think we should start from the new_food_path here, then test if filling in works or not.

You should also test what happens when the user doesn't fill every necessary field

Reply:
good catch.